### PR TITLE
autostart: distinguish between init and upstart

### PR DESCRIPTION
**Pull request by brammeleman**
**Description:**
Added upstart configuration + documentation

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant):** home-assistant/home-assistant#4243

### CODE REVIEW

File: source/getting-started/autostart-init.markdown [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+To install this script, download it, tweak it to you liking, and install it by following the directions in the header. This script will setup Home Assistant to run when the system boots. To make sure the daemon runs at the run levels specified in the header of the script, run the folowing command:
+
+```
+$ sudo update-rc.d hass-daemon defaults
+```
+
+This will create the appropiate symlinks in the /etc/rc${runlevel}.d/ directories on Debian based systems. For Red Hat derivatives, use `chkconfig` instead:
+
+```
```
Comment (Landrash) [inline source/getting-started/autostart-init.markdown]: Change this to 
\`\`\`bash
to get the appropriate highlight of the code.

File: source/getting-started/autostart-upstart.markdown [hunk 2]
```diff
@@ -9,23 +9,47 @@ sharing: true
...
 
 ```bash
 $ ps -p 1 -o comm=
 ```
 
-If the preceding command returns the string `init`, you are likely using Upstart.
+If the preceding command returns the string `init`, you are likely using Upstart. Just to be really sure, request the Upstart version with:
 
-Upstart will launch init scripts that are located in the directory `/etc/init.d/`. A sample init script for systems using Upstart is <a href="https://raw.githubusercontent.com/home-assistant/home-assistant/dev/script/hass-daemon">maintained by this project</a>.
+```
```
Comment (Landrash) [inline source/getting-started/autostart-upstart.markdown]: Change this to 
\`\`\`bash
to get the appropriate highlight of the code.

File: source/getting-started/autostart-upstart.markdown [hunk 3]
```diff
@@ -9,23 +9,47 @@ sharing: true
...
+$ init --version
+```
+
+The output should mention something similar to `init (upstart 1.12.1)`. If it doesn't, you're dealing with a traditional (SysV-style) init system and you should follow [these instructions](../autostart-init).
```
Comment (Landrash) [inline source/getting-started/autostart-upstart.markdown]: Link should be to `/getting-started/autostart-init/`.

File: source/getting-started/autostart-upstart.markdown [hunk 4]
```diff
@@ -9,23 +9,47 @@ sharing: true
...
+
+Upstart will use configuration files in `/etc/init/`. A sample Upstart configuration is included in the <a href="https://github.com/home-assistant/home-assistant/blob/dev/script/hass.conf">Home Assistant source code</a>.
+
+Installation
+-------------
+Create a file named `/etc/init/hass.conf` with the following contents:
+
+```
```
Comment (Landrash) [inline source/getting-started/autostart-upstart.markdown]: Change this to 
\`\`\`bash
to get the appropriate syntax highlighting.

File: source/getting-started/autostart-upstart.markdown [hunk 5]
```diff
@@ -9,23 +9,47 @@ sharing: true
...
+start on runlevel [2345]
+stop on runlevel [016]
+
+setuid hass
+
+# change paths according to your situation:
+exec /srv/hass/bin/python3 /srv/hass/bin/hass -c /home/hass/homeassistant --pid-file /home/hass/homeassistant/hass.pid
+```
+
+Be sure to check the paths in the last line. That is where it all gets toghether. That's it! From now on, Upstart will make sure Home Assistent is started at boot and stopped when shutting down the system. To start Home Assistent manually, use:
```
Comment (Landrash) [inline source/getting-started/autostart-upstart.markdown]: Home Assistent should be Home Assistant.

File: source/getting-started/autostart-upstart.markdown (hunk 4 above)
Comment (brammeleman) [inline source/getting-started/autostart-upstart.markdown]: Bash syntax highlighting does not apply here.

Comment (Landrash) [inline source/getting-started/autostart-upstart.markdown]: Sorry, your correct ofc. It should be `shell`.

Good reference list [here][syntaxreference] if you use markdown with jekyll or Github.

[syntaxreference]:https://github.com/github/linguist/blob/master/lib/linguist/languages.yml#L3957

Comment (brammeleman) [inline source/getting-started/autostart-upstart.markdown]: It isn't shell either, it's an upstart configuration file. So I think general code block would apply.

**Comment by Landrash** [comment]
@brammeleman Would you mind fixing the merge conflict? After that this is good to go.

**Comment by brammeleman** [comment]
There's no merge conflig. Is it?

**Comment by Landrash** [comment]
There is actually. @fabaff would you mind taking care of it?

**Comment by brammeleman** [comment]
Ah, now I see. I'll have a look.
/Bram

**Comment by brammeleman** [comment]
whoopsy... 
Running into serious git problems when trying to fetch upstream changes.

**Comment by brammeleman** [comment]
Sorry for the mess. Mixing up the mater and next branch. If you whish, I'll start all over. Just let me know.
/B

**Comment by andrey-git** [comment]
This branch seems broken. Please make a new PR

**Pull request closed without merging**